app/javascript/src/components/actions/submittable.js

```diff
@@ -0,0 +1,26 @@
+
+export function submittable(node) {
+  const nodeIsInput = node.tagName === "INPUT"
+  const nodeIsTextarea = node.tagName === "TEXTAREA"
+
+  if (!(nodeIsInput || nodeIsTextarea)) {
+    throw new Error("submittable action is only meant to be used on inputs")
+  }
```

Comment:
I don't think think textarea should be included, the default behaviour of enter on textarea should still be new lines, as that is expected. If you don't know you will accidentally submit too early.
```suggestion

  if (!nodeIsInput) {
    throw new Error("submittable action is only meant to be used on inputs")
  }
```

Reply:
How about allow this in textareas but only when pressing Ctrl + Enter?

Reply:
Sure, I like that!